ddtrace/appsec/_iast/_taint_tracking/taint_tracking/taint_range.cpp

```diff
@@ -266,6 +193,14 @@ get_ranges(PyObject* string_input, const TaintedObjectMapTypePtr& tx_map)
 bool
 set_ranges(PyObject* str, const TaintRangeRefs& ranges, const TaintedObjectMapTypePtr& tx_map)
 {
+    // Guard: invalid taint map
+    if (not tx_map) {
+        return false;
+    }
+    // Guard: only text-like objects are supported for taint
```

Comment:
Maybe this could use conditional compilation to elide the check when compiled in non debug mode (I saw you had to increase the times in the benchmarks and I suspect this could be the cause).